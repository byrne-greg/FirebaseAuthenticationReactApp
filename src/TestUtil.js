import React from "react";
import { MemoryRouter } from "react-router-dom";

const withTestRouter = Component => <MemoryRouter>{Component}</MemoryRouter>;

export { withTestRouter };
