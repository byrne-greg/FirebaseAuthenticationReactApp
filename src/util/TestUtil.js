import React from "react";
import { MemoryRouter } from "react-router-dom";
import { FirebaseContext } from "../components/Firebase";
import { AuthUserContext } from "../components/Session";

const withMockFirebase = (mockFirebase, Component) => (
  <FirebaseContext.Provider value={mockFirebase}>
    {Component}
  </FirebaseContext.Provider>
);

const withMockAuthUserContext = (value, Component) => (
  <AuthUserContext.Provider value={value}>{Component}</AuthUserContext.Provider>
);

const withTestRouter = Component => <MemoryRouter>{Component}</MemoryRouter>;

export { withTestRouter, withMockFirebase, withMockAuthUserContext };
