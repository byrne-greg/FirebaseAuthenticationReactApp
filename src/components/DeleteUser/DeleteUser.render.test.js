import React from "react";
import { render, cleanup } from "@testing-library/react";
import { withTestRouter } from "../../util/TestUtil";
import { DeleteUserForm, DeleteUserSuccessfulPage } from ".";

afterEach(cleanup);

describe("DeleteUser render tests", () => {
  it("delete user form renders as expected", () => {
    // GIVEN

    // WHEN
    const form = render(withTestRouter(<DeleteUserForm />));

    // THEN
    expect(form.baseElement).toMatchSnapshot();
  });

  it("delete user form renders as expected", () => {
    // GIVEN

    // WHEN
    const page = render(withTestRouter(<DeleteUserSuccessfulPage />));

    // THEN
    expect(page.baseElement).toMatchSnapshot();
  });
});
