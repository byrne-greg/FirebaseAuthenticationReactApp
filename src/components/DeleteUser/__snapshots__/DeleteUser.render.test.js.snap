// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteUser render tests delete user form renders as expected 1`] = `
<body>
  <div>
    <h2>
      Delete your account
    </h2>
    <form>
      <label
        id="confirm-delete-user-label"
      >
        Are you sure you want to delete your account?
      </label>
      <input
        aria-labelledby="confirm-delete-user-label"
        name="confirm-delete"
        type="checkbox"
        value="false"
      />
      <button
        disabled=""
        id="submit-button"
        type="submit"
      >
        Delete account
      </button>
    </form>
  </div>
</body>
`;
